### ISSUE TITLE
diyan的dloss的计算

### ISSUE THREAD

**Issue by kyle141377** [JiahengHu/DUSDi#2]
您好，我注意到在dusdi_diyan.py中的第291行，d_loss = d_loss + d_anti_loss。从文中的思路来看这一步是实现对skill的预测仅依赖于channel内部（由d_loss决定）而与channel外部（由d_anti_loss决定）的dim解耦的关键步骤。直观的理解，这里应该是d_loss = d_loss - d_anti_loss，请问代码中为什么采用相加呢？

**Comment by kyle141377** [comment]
Hello, I noticed that in dusdi_diayn.py at line 291, there is d_loss = d_loss + d_anti_loss. According to the idea described in the paper, this step is crucial for ensuring that the prediction of the skill depends only on the dimensions within the channel (determined by d_loss) and is decoupled from the dimensions outside the channel (determined by d_anti_loss). Intuitively, it seems that this should be d_loss = d_loss - d_anti_loss. Why does the code use addition here?

**Comment by JiahengHu** [comment]
Hi. Thanks for asking! The loss here is for updating the diayn (and anti_diayn) predictor. So both of them have the same sign. In line 228, you will notice that the sign is changed when computing the intrinsic reward. Let me know if you have more questions. Thank you

**Comment by kyle141377** [comment]
> Hi. Thanks for asking! The loss here is for updating the diayn (and anti_diayn) predictor. So both of them have the same sign. In line 228, you will notice that the sign is changed when computing the intrinsic reward. Let me know if you have more questions. Thank you

thanks a lot, i had noticed the calculate of reward in line [228.is](http://228.is/) it necessary to make anti_diayn predict the skill accurate in line292, and then to update the reward in line?

**Comment by JiahengHu** [comment]
Correct. Predicting the skill as accurately as possible is necessary for approximating the mutual information
